config/filter.d/asterisk.conf

```diff
@@ -25,14 +25,24 @@ failregex = ^(%(__prefix_line)s|\[\]\s*)%(log_prefix)s Registration from '[^']*'
             ^(%(__prefix_line)s|\[\]\s*)%(log_prefix)s Host <HOST> failed MD5 authentication for '[^']*' \([^)]+\)$
             ^(%(__prefix_line)s|\[\]\s*)%(log_prefix)s Failed to authenticate (user|device) [^@]+@<HOST>\S*$
             ^(%(__prefix_line)s|\[\]\s*)%(log_prefix)s hacking attempt detected '<HOST>'$
-            ^(%(__prefix_line)s|\[\]\s*)%(log_prefix)s SecurityEvent="(FailedACL|InvalidAccountID|ChallengeResponseFailed|InvalidPassword)",EventTV="([\d-]+|%(iso8601)s)",Severity="[\w]+",Service="[\w]+",EventVersion="\d+",AccountID="(\d*|<unknown>)",SessionID=".+",LocalAddress="IPV[46]/(UDP|TCP|WS)/[\da-fA-F:.]+/\d+",RemoteAddress="IPV[46]/(UDP|TCP|WS)/<HOST>/\d+"(,Challenge="[\w/]+")?(,ReceivedChallenge="\w+")?(,Response="\w+",ExpectedResponse="\w*")?(,ReceivedHash="[\da-f]+")?(,ACLName="\w+")?$
+            ^(%(__prefix_line)s|\[\]\s*)%(log_prefix)s <HOST> tried to authenticate with nonexistent user.+$
+            ^(%(__prefix_line)s|\[\]\s*)%(log_prefix)s <HOST> failed to authenticate as.+$
```

Comment:
I said the regexp `.+$` makes at all no sense. And I mean it's really so. What is there not to understand? 
Just compare:
```bash
python -c 'import re; print(True if re.search(r"^XYZ\s",  "XYZ no matter what is here") else False)'
True
python -c 'import re; print(True if re.search(r"^XYZ.+$", "XYZ no matter what is here") else False)'
True
python -c 'import re; print(True if re.search(r"^XYZ\s",  "XYZ ") else False)'
True
python -c 'import re; print(True if re.search(r"^XYZ.+$", "XYZ ") else False)'
True
```
As regards `o'reilley`, did you see the anchor `$` was removed, and my expression was `\s'[^']+'`
```bash
fail2ban-regex \
"[2016-05-06 07:08:09] NOTICE[6772] manager.c: 1.2.3.4 failed to authenticate as 'o'reilley'" \
"^(?:\[\])?\s*\S+\[\d+\] \S+ <HOST> failed to authenticate as '[^']'"
```
But I'm against any anchored almost catch-alls, that are not clear defined, so just replace it gegen `\s`.
```bash
fail2ban-regex \
"[2016-05-06 07:08:09] NOTICE[6772] manager.c: 1.2.3.4 failed to authenticate as 'o'reilley'" \
"^(?:\[\])?\s*\S+\[\d+\] \S+ <HOST> failed to authenticate as\s"

Lines: 1 lines, 0 ignored, 1 matched, 0 missed
[processed in 0.01 sec]
```